code/modules/multiz/pipes.dm

```diff
@@ -43,6 +48,20 @@
 		if(SOUTHWEST)
 			initialize_directions = SOUTH
 
+/obj/machinery/atmospherics/pipe/zpipe/Entered(atom/movable/Obj)
+	if(istype(Obj, /mob/living))
+		var/mob/living/L = Obj
+		L << span("notice", "You are in a vertical pipe section. Use <a href='?src=\ref[src];crawl_user=\ref[L];crawl_dir=[travel_direction]'>[travel_verbname]</a> from the IC menu to [travel_direction_verb] a level.")
+	. = ..()
+
+/obj/machinery/atmospherics/pipe/zpipe/Topic(href, href_list)
+	.=..()
+	if (href_list["crawl_user"])
+		var/mob/living/L = locate(href_list["crawl_user"])
```

Comment:
I would validate L here to make sure that they are in-fact proximal to the pipe; Topic() arguments cannot be trusted to be valid or sane.